Segmentation fault (Linux)

### Description

```
Traceback (most recent call last)
/home/anastasiya/status-desktop/src/nim_status_client.nim(208) nim_status_client
/home/anastasiya/status-desktop/src/nim_status_client.nim(181) mainProc
/home/anastasiya/status-desktop/src/app/boot/app_controller.nim(297) delete
/home/anastasiya/status-desktop/src/app/modules/startup/io_interface.nim(19) delete
/home/anastasiya/status-desktop/src/app/modules/startup/module.nim(87) delete
/home/anastasiya/status-desktop/vendor/nimqml/src/nimqml/private/qqmlapplicationengine.nim(29) setRootContextProperty
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
make: *** [Makefile:815: run-linux] Segmentation fault (core dumped)
```

1. build the app on linux (latest master)
2. close the app by clicking cross button
3. check the console

--- a/src/app/boot/app_controller.nim
+++ b/src/app/boot/app_controller.nim
@@ -477,7 +477,7 @@ proc finishAppLoading*(self: AppController) =
     self.privacyService.removeMnemonic()
 
   if not self.startupModule.isNil:
-    self.startupModule.delete
+    self.startupModule.onAppLoaded()
     self.startupModule = nil
 
   self.mainModule.checkAndPerformProfileMigrationIfNeeded()

--- a/src/app/modules/main/wallet_section/send/account_item.nim
+++ b/src/app/modules/main/wallet_section/send/account_item.nim
@@ -44,7 +44,7 @@ QtObject:
     self.canSend = canSend
 
   proc delete*(self: AccountItem) =
-      self.QObject.delete
+    self.QObject.delete
 
   proc newAccountItem*(
     name: string = "",

--- a/src/app/modules/main/wallet_section/send/view.nim
+++ b/src/app/modules/main/wallet_section/send/view.nim
@@ -35,11 +35,13 @@ QtObject:
   proc delete*(self: View) =
     self.accounts.delete
     self.senderAccounts.delete
-    self.selectedSenderAccount.delete
+    if self.selectedSenderAccount != nil:
+      self.selectedSenderAccount.delete
     self.fromNetworksModel.delete
     self.toNetworksModel.delete
     self.transactionRoutes.delete
-    self.selectedReceiveAccount.delete
+    if self.selectedReceiveAccount != nil:
+      self.selectedReceiveAccount.delete
     self.QObject.delete
 
   proc newView*(delegate: io_interface.AccessInterface): View =

--- a/src/app/modules/startup/io_interface.nim
+++ b/src/app/modules/startup/io_interface.nim
@@ -19,6 +19,9 @@ type
 method delete*(self: AccessInterface) {.base.} =
   raise newException(ValueError, "No implementation available")
 
+method onAppLoaded*(self: AccessInterface) {.base.} =
+  raise newException(ValueError, "No implementation available")
+
 method load*(self: AccessInterface) {.base.} =
   raise newException(ValueError, "No implementation available")
 

--- a/src/app/modules/startup/module.nim
+++ b/src/app/modules/startup/module.nim
@@ -84,6 +84,13 @@ proc newModule*[T](delegate: T,
 {.push warning[Deprecated]: off.}
 
 method delete*[T](self: Module[T]) =
+  self.view.delete
+  self.viewVariant.delete
+  self.controller.delete
+  if not self.keycardSharedModule.isNil:
+    self.keycardSharedModule.delete
+
+method onAppLoaded*[T](self: Module[T]) =
   singletonInstance.engine.setRootContextProperty("startupModule", newQVariant())
   self.view.delete
   self.view = nil

